app/kubernetes/views/applications/stats/statsController.js

```diff
@@ -0,0 +1,161 @@
+import angular from 'angular';
+import moment from 'moment';
+import _ from 'lodash-es';
+import filesizeParser from 'filesize-parser';
+import KubernetesResourceReservationHelper from 'Kubernetes/helpers/resourceReservationHelper';
+
+class KubernetesApplicationStatsController {
+  /* @ngInject */
+  constructor($async, $state, $interval, $document, Notifications, KubernetesPodService, KubernetesNodeService, KubernetesMetricsService, ChartService) {
+    this.$async = $async;
+    this.$state = $state;
+    this.$interval = $interval;
+    this.$document = $document;
+    this.Notifications = Notifications;
+    this.KubernetesPodService = KubernetesPodService;
+    this.KubernetesNodeService = KubernetesNodeService;
+    this.KubernetesMetricsService = KubernetesMetricsService;
+    this.ChartService = ChartService;
+
+    this.onInit = this.onInit.bind(this);
```

Comment:
shall we bind other functions that uses "this"?

Reply:
Only when it is async functions (https://github.com/portainer/portainer/pull/2945)